File: esthesis-platform-server/src/main/java/esthesis/platform/server/dto/ContainerDTO.java
package esthesis.platform.server.dto;

import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.experimental.Accessors;

import java.util.List;

@Data
@NoArgsConstructor
@Accessors(chain = true)
public class ContainerDTO {
  private String image;
  private String registryUsername;
  private String registryPassword;
  private String network;
  private String restart;
  private List<ContainerPortDTO> ports;
  private List<ContainerVolumeDTO> volumes;
  private List<ContainerEnvDTO> env;
  private int server;
  private String name;

  // The number of replicas.
  private int scale;

  // The WebSocket topic id to send progress messages.
  private String wsId;
}

File: esthesis-platform-ui/src/app/dto/container-dto.ts
import {ContainerPortDto} from './container-port-dto';
import {ContainerVolumeDto} from './container-volume-dto';
import {ContainerEnvDto} from './container-env-dto';

export class ContainerDto {
  container: string;
  registryUsername: string;
  registryPassword: string;
  network: string;
  restart: string;
  ports: ContainerPortDto[];
  volumes: ContainerVolumeDto[];
  env: ContainerEnvDto[];
  scale: number;
}
